helm/vmss-prototype/README.md

```diff
@@ -126,8 +115,58 @@ The following Helm Chart values are exposed to configure a `vmss-prototype` rele
   - e.g., `--set kamino.drain.force=true`
   - Override the default if you wish to force a drain even if the grace period has expired, or if there are non-replicated pods (e.g., not part of a ReplicationController, ReplicaSet [i.e., Deployment], DaemonSet, StatefulSet or Job). This is directly equivalent to the `--force` flag of the `kubectl drain` CLI operation
 
-More documentation on node drain can be found [here in the `kubectl` documentation](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#drain).
+  More documentation on node drain can be found [here in the `kubectl` documentation](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#drain).
+
 
 - `kamino.logLevel` (default value is `INFO`, options are `DEBUG`, `INFO`, `WARNING`, `ERROR`, and `CRITICAL`)
-  - e.g., `--set kamino.logLevel=VERBOSE`
+  - e.g., `--set kamino.logLevel=DEBUG`
   - Override the default if you wish to see more or less log output in the `vmss-prototype` pod.
+
+### Automatic node selection
+
+Kamino can follow some rules for automatic node selection.  These are the parameters for the helm chart (in addition to the above) to enable this.
+
+- `kamino.targetVMSS` (required to run in automatic mode)
+  - A value of `ALL` will automatically scan for all VMSS pools
+    - e.g., `--set kamino.targetVMSS=ALL`
+  - A value of the VMSS name.  This is the name of the VMSS without the node-specific identifier appended.  For example, `k8s-pool1-12345678-vmss` would be the VMSS that contains nodes such as `k8s-pool1-12345678-vmss000000`
+    - e.g., `--set kamino.targetVMSS=k8s-pool1-12345678-vmss`
+
+- `kamino.auto.lastPatchAnnotation` (required)
+  - This defines the name of the annotation on the nodes that holds an RFC3339 timestamp of when it last applied an OS patch.  If this annotation is missing, it is assumed the node has not applied an OS patch yet.
+    - e.g., `--set kamino.auto.lastPatchAnnotation=LatestOSPatch`
+
+- `kamino.auto.pendingRebootAnnotation` (required ??)
+  - This annotation on a node that indicates that the node is in need of reboot for some reason (pending OS patch or other servicing).  Nodes with this annotation are not considered as a viable candidate for making an image.
+    - e.g., `--set kamino.auto.pendingRebootAnnotation=RebootPending`
+  - Note that setting this to some annotation that never exists causes the system to not filter out any nodes due to this constraint.
+
+- `kamino.auto.minimumReadyTime` (default 1h)
+  - This is the minimum time a node must have been in constant "ready" state for it to be considered a potential candidate as the prototype.

```

Comment:
I think the "Ready" state of a Node and its age are not directly correlated. For example, if a node flaps between Ready and NotReady I don't think the age of node is reset to zero every time it reconciles back to "Ready".

So if that's correct, then I think we need to say something like "This is the minimum age of a node for it to be considered a potential candidate as the prototype.

Reply:
No, the age of the node is not reset - but the transition time to ready is stored.   The state we look at is the time it last transitioned into ready.  That is part of the kubernetes node data that "kubectl get nodes -o json" will produce.

Reply:
As such, the documentation is correct.  It is not node age but continuous time in ready (time since last transition to ready state)